imap_data_access/cli.py

```diff
@@ -98,14 +106,50 @@ def _query_parser(args: argparse.Namespace):
         "repointing",
         "version",
         "extension",
+        "filename",
     ]
+    latest_version = ""
     query_params = {
         key: value
         for key, value in vars(args).items()
         if key in valid_args and value is not None
     }
+
+    # Checking to see if a filename was passed.
+    if args.filename is not None:
+        if len(sys.argv) > 4:  # I have doubts about this line
+            raise TypeError("Too many arguments, '--filename' should be ran by itself")
```

Comment:
To address your doubts - this seems ok to me. You could also check to see if "instrument", "data_level", etc is in sys.argv, which would be more reliable in case we add some additional options which _can_ be passed in alongside filename.

Reply:
Actually, you could also validate that `query_params` is empty, since we would expect it to be empty if `filename` was passed in, and we're rewriting it anyway so it's a good thing to check